### PR TITLE
allow use a separate pauser instead of global pauser

### PR DESCRIPTION
**Pull request by bobotu**
### What problem does this PR solve? 
Allow pass a pauser to `RestoreController` to pause and resume a single `RestoreController`

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: lightning/restore/restore.go [hunk 1]
```diff
@@ -153,6 +154,10 @@ type RestoreController struct {
 }
 
 func NewRestoreController(ctx context.Context, dbMetas []*mydump.MDDatabaseMeta, cfg *config.Config) (*RestoreController, error) {
+	return NewRestoreControllerWithPauser(ctx, dbMetas, cfg, DeliverPauser)
```
Comment (suzaku) [inline lightning/restore/restore.go]: Shall we rename `DeliverPauser` as `DefaultPauser`?  cc @kennytm

Comment (kennytm) [reply lightning/restore/restore.go]: Either is fine.

Comment (suzaku) [reply lightning/restore/restore.go]: Do you know why the test remains "Pending" for hours?

Comment (kennytm) [reply lightning/restore/restore.go]: Jenkins config issue. Fixing it @suzaku

**Comment by kennytm** [comment]
Please check the unit test failure @bobotu, thanks.

**Comment by bobotu** [comment]
@suzaku PTAL

**Pull request merged**